packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -392,35 +385,26 @@ export default class ScriptTransformer {
   private _transformAndBuildScript(
     filename: Config.Path,
     options: Options,
-    instrument: boolean,
-    fileSource?: string,
+    transformOptions: TransformOptions,
+    fileSource: string,
   ): TransformResult {
-    const {
-      isCoreModule,
-      isInternalModule,
-      supportsDynamicImport,
-      supportsStaticESM,
-    } = options;
-    const content = stripShebang(
-      fileSource || fs.readFileSync(filename, 'utf8'),
```

Comment:
runtime already sends in the source, seemed like a good time to remove this fallback

Reply:
ugh, too painful to fix the tests, can circle back to it later. I pushed WIP stuff here: https://github.com/SimenB/jest/commit/31d67e229f8595def36dd88fa5745a11e9ad79b7